### PR TITLE
Add Rbac rules for new MariaDB service

### PR DESCRIPTION
**Pull request by TheBigLee**
## Summary

* This PR adds the required RBAC rules in order to be able to access the created service

## Checklist

- [x] Categorize the PR by setting a good title and adding one of the labels:
      `bug`, `enhancement`, `documentation`, `change`, `breaking`, `dependency`
      as they show up in the changelog
- [x] Update tests.
- [ ] Link this PR to related issues.

### CODE REVIEW

**Review by Kidswiss — CHANGES_REQUESTED** [review]
Some input to reduce the complexity and make it a bit more efficient.

File: pkg/comp-functions/functions/common/namespace-permissions.go [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+		ObjectMeta: metav1.ObjectMeta{
+			Name: claimNs,
+		},
+	}
+
+	return svc.SetDesiredKubeObserveObject(ns, instance+"-claim-ns-observer")
+}
+
+func CreateNamespacePermissions(ctx context.Context, instance string, svc *runtime.ServiceRuntime) error {
+	ns := &corev1.Namespace{}
```
[the remark this answers is not in the document]
Comment (TheBigLee) [reply pkg/comp-functions/functions/common/namespace-permissions.go:35]: We need this to get the `organizationLabel` from the claimNamespace

File: pkg/comp-functions/functions/common/namespace-permissions.go [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+			return fmt.Errorf("cannot get namespace: %w", err)
+		}
+	}
+	instanceNs := ns.GetName()
+
+	org := ns.GetLabels()[utils.OrgLabelName]
+
+	roleBinding := &rbacv1.RoleBinding{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "appcat:services:read",
```
[the remark this answers is not in the document]
Comment (TheBigLee) [reply pkg/comp-functions/functions/common/namespace-permissions.go]: Fixed

File: pkg/comp-functions/functions/common/namespace-permissions.go [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+			Name: claimNs,
+		},
+	}
+
+	return svc.SetDesiredKubeObserveObject(ns, instance+"-claim-ns-observer")
+}
+
+func CreateNamespacePermissions(ctx context.Context, instance string, svc *runtime.ServiceRuntime) error {
+	ns := &corev1.Namespace{}
+	err := svc.GetDesiredKubeObject(ns, instance+"-ns")
```
[the remark this answers is not in the document]
Comment (TheBigLee) [reply pkg/comp-functions/functions/common/namespace-permissions.go]: Yes. Fixed it

File: pkg/comp-functions/functions/common/namespace-permissions.go [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+	rbacv1 "k8s.io/api/rbac/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const roleBindingName = "appcat:services:read"
+
+func GetOrg(instance string, svc *runtime.ServiceRuntime) string {
+	ns := &corev1.Namespace{}
+
+	err := svc.GetObservedKubeObject(ns, instance+"-claim-ns-observer")
```
Comment (zugao) [inline pkg/comp-functions/functions/common/namespace-permissions.go:19]: I don't like the fact that we are hard coding parts of the object names. There's no other way to get the namespace?

Comment (TheBigLee) [reply pkg/comp-functions/functions/common/namespace-permissions.go]: I've moved this into a const. Is it accetable for you now?

Comment (zugao) [reply pkg/comp-functions/functions/common/namespace-permissions.go]: I meant something else but it's fine. We need a better way to managed names in our comp functions, pre pending suffixes and prefixes all over is gonna cost us dearly later on.

**Review by zugao — APPROVED** [review]

**Review by Kidswiss — APPROVED** [review]

**Pull request merged**